### PR TITLE
Ignore Envelopes with Unkown Validators

### LINKED ISSUE #544: Change checks for unknown validators in Stellar relay pallet
## Context 

The stellar relay pallet currently rejects the whole 'proof' if one of the passed SCP envelopes was signed by an unknown validator or if any other related check fails, see [here](https://github.com/pendulum-chain/spacewalk/blob/d74009bf1d8c27fbd1f314c4fe486cd230e8b73f/pallets/stellar-relay/src/validation.rs#L163-L189). However, this check is a little too strict as it might happen that a set of envelopes can still be used to build a valid proof, even if some of the contained envelopes are 'invalid'.

## TODO

Instead of returning an error, we should rather change the logic so that the [validate_envelopes](https://github.com/pendulum-chain/spacewalk/blob/d74009bf1d8c27fbd1f314c4fe486cd230e8b73f/pallets/stellar-relay/src/validation.rs#L152) function checks each envelope for validity but only _returns the valid envelopes_. These valid envelopes can then be passed to the [check_for_valid_quorum_set](https://github.com/pendulum-chain/spacewalk/blob/d74009bf1d8c27fbd1f314c4fe486cd230e8b73f/pallets/stellar-relay/src/lib.rs#L589) function to continue with the existing validity checks.

### PR DESCRIPTION
**Pull request by b-yap**
closes #544 

Need to close https://github.com/pendulum-chain/substrate-stellar-sdk/pull/32 before proceeding.

This will ignore validating envelopes with a different `node_id`.

### CODE REVIEW

**Review by ebma — COMMENTED** [review]
The logic looks good to me in general. We also need to fix the CI errors though.

**Comment by b-yap** [comment]
https://github.com/pendulum-chain/spacewalk/actions/runs/10263219351/job/28396526378?pr=546#step:13:556
Tested in my local machine, and works with Iowa and Singapore. 

Frankfurt on the otherhand:
Screenshot 2024-08-06 at 7 26 34 PM
```
[2024-08-06T11:27:21Z ERROR stellar_relay_lib::connection::connector::message_handler] process_raw_message(): Received ErrorMsg during authentication: Error{ code:ErrLoad message:peer rejected }
[2024-08-06T11:27:21Z ERROR stellar_relay_lib::connection::error] Stellar Node returned error: Error{ code:ErrLoad message:peer rejected }
```

I'm [removing Frankfurt in the list](https://github.com/pendulum-chain/spacewalk/pull/546/commits/15d873da98085e69c40dbd6d60586684e9c2f33a), for the meantime.

File: .github/workflows/ci-main.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ name: continuous-integration-main
 jobs:
   ci:
     strategy:
-      max-parallel: 2
+      max-parallel: 1
```
Comment (ebma) [inline .github/workflows/ci-main.yml:10]: Should we turn this back again?

File: pallets/stellar-relay/src/validation.rs [hunk 2]
```diff
@@ -187,22 +202,13 @@ pub fn validate_envelopes<T: Config>(
...
-) -> Result<&ScpEnvelope, Error<T>> {
-	envelopes
-		.get_vec()
-		.iter()
-		.find(|envelope| match envelope.statement.pledges {
-			ScpStatementPledges::ScpStExternalize(_) => true,
-			_ => false,
-		})
-		.ok_or(Error::<T>::MissingExternalizedMessage)
+	// it's ok to use unwrap here, since the size will be <= to the provided envelopes
```
Comment (ebma) [inline pallets/stellar-relay/src/validation.rs:210]: Nit: remove stale comment

**Review by ebma — APPROVED** [review]

**Comment by b-yap** [comment]
@ebma after I fix the fmt issue, I'll merge it immediately.

**Pull request merged**